funcbench/main.go

```diff
@@ -285,33 +289,14 @@ func interrupt(logger Logger, cancel <-chan struct{}) error {
 	}
 }
 
-// getTargetInfo returns the hash of the target,
-// if target is the same as the current ref, set compareWithItself to true.
-func getTargetInfo(ctx context.Context, repo *git.Repository, target string) (ref plumbing.Hash, compareWithItself bool, _ error) {
-	if target == "." {
-		return plumbing.Hash{}, true, nil
-	}
-
-	currRef, err := repo.Head()
+// getTargetInfo returns the hash of the target if found,
+// otherwise returns plumbing.ZeroHash.
+func getTargetInfo(repo *git.Repository, target string) plumbing.Hash {
+	hash, err := repo.ResolveRevision(plumbing.Revision(target))
```

Comment:
Great improvement! :partying_face: thanks for looking this up! might still want to add a todo note about the same target name for `tag` and `branch`, I tried running it a couple of times, it's always choosing the branch commit hash which can someday maybe change in go-git, so just a small note about it will be helpful. Think ready to merge after that :)